rhombus/private/list.rkt

```diff
@@ -621,6 +622,12 @@
   #:primitive (treelist-ref)
   (treelist-ref l n))
 
+(define/method (List.set l n v)
```

Comment:
Should we call this `List.set_in_copy`, like `Array.set_in_copy`, to avoid suggesting `set!` as opposed to `set` (and thus avoid suggesting `MutableIndexable`)?

Reply:
I think we should reserve "copy" for creating mutable copies. For now, I've left this as `set`, but we can discussion more what the right name should be.